packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/pools/_components/formPools/formConnectionPool.hook.tsx

```diff
@@ -0,0 +1,95 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { database } from '@/models/database';
+import { POOL_CONFIG } from './connectionPool.const';
+
+export interface UseConnectionPoolFormProps {
+  existingConnectionPools?: database.postgresql.ConnectionPool[];
+  editedConnectionPool?: database.postgresql.ConnectionPool;
+}
+export const useConnectionPoolForm = ({
+  existingConnectionPools,
+  editedConnectionPool,
+}: UseConnectionPoolFormProps) => {
+  const { t } = useTranslation(
+    'pci-databases-analytics/services/service/pools',
+  );
+
+  const usedNames = editedConnectionPool
+    ? []
+    : existingConnectionPools.map((cp) => cp.name);
```

Comment:
```suggestion
  const usedNames = existingConnectionPools.filter(c => c.name !== editedConnectionPools?.name).map((cp) => cp.name);
```
By filtering the name of the editing pool, you should not have errors during the edition. You could remove the '?' in the props then